dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -2017,8 +2020,29 @@ void DeltaMergeStore::applySchemaChanges(TiDB::TableInfo & table_info)
     original_table_columns.swap(new_original_table_columns);
     store_columns.swap(new_store_columns);
 
-    // TODO(local index): There could be some local indexes added/dropped after DDL
+    // copy the local_index_infos to check whether any new index is created
+    LocalIndexInfosPtr local_index_infos_copy = nullptr;
+    {
+        std::shared_lock index_read_lock(mtx_local_index_infos);
+        local_index_infos_copy = std::shared_ptr<LocalIndexInfos>(local_index_infos);
+    }
+
     std::atomic_store(&original_table_header, std::make_shared<Block>(toEmptyBlock(original_table_columns)));
+
+    // release the lock because `applyLocalIndexChange` will try to acquire the lock
```

Comment:
Do we need to update `applyLocalIndexChange` in this comment?

Reply:
fixed